### PR TITLE
Add pre-commit-hooks to prepare for  hlint and clean up format checking

### PR DESCRIPTION
**Pull request by MangoIV**
- removed hacky formatchecking 
- prepared for running hlint check as soon as we have all the hints applied

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -1,44 +1,72 @@
...
-  inputs.protolude.flake = false;
-
-  # 0.4.5 hasn't been published to Hackage...
-  inputs.flat.url = "github:Quid2/flat";
-  inputs.flat.flake = false;
-
-  inputs.emanote.url = "github:srid/emanote/master";
-
-  outputs = inputs@{ self, nixpkgs, nixpkgs-latest, iohk-nix, haskell-nix, hercules-ci-effects, haskell-nix-extra-hackage, ... }:
+  inputs = {
```
Comment (L-as) [inline flake.nix:4]: ?

File: plutarch-test/Main.hs [hunk 2]
```diff
@@ -38,6 +38,7 @@ spec :: Spec
 spec = do
   BaseSpec.spec
   ExtraSpec.spec
+
```
Comment (L-as) [inline plutarch-test/Main.hs:41]: Please don't include irrelevant changes

File: flake.nix (hunk 1 above)
Comment (MangoIV) [reply flake.nix:4]: especially as it gets more. it's also less to type.

Comment (L-as) [reply flake.nix:4]: You have more than double the lines in an already long file.

**Comment by L-as** [comment]
How do you format the code?

**Comment by L-as** [comment]
How is this better than what we have before?

**Comment by MangoIV** [comment]
- It's more readable
- easier to use
- will make adding more checks absolutely easy 
- most importantly: runs automatically so you never have an unformatted commit

**Comment by L-as** [comment]
I've decided against this unfortunately. While the idea is nice, pre-commit is pretty much just bloatware.
The only useful feature is that it automatically adds the git commit hook when you enter a shell. That's trivial to reimplement. It is not at all worth the complexity of this *huge* framework. It would take ~20 lines of bash to reimplement.

**Pull request closed without merging**